service_container/tags.rst

```diff
@@ -564,3 +564,216 @@ application handlers.
+but we can now access a specific injected service:
+
+.. configuration-block::
+
+    .. code-block:: yaml
+
+        # config/services.yaml
+        services:
+            App\Handler\One:
+                tags:
+                    - { name: 'app.handler', key: 'handler_one' }
+
+            App\Handler\Two:
+                tags:
+                    - { name: 'app.handler', key: 'handler_two' }
+
+            App\HandlerCollection:
+                # inject all services tagged with app.handler as first argument
+                arguments: [!tagged { tag: 'app.handler', index_by: 'key' }]
+
+    .. code-block:: xml
+
+        <!-- config/services.xml -->
+        <?xml version="1.0" encoding="UTF-8" ?>
+        <container xmlns="http://symfony.com/schema/dic/services"
+            xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+            xsi:schemaLocation="http://symfony.com/schema/dic/services
+                http://symfony.com/schema/dic/services/services-1.0.xsd">
+
+            <services>
+                <service id="App\Handler\One">
+                    <tag name="app.handler" key="handler_one" />
+                </service>
+
+                <service id="App\Handler\Two">
+                    <tag name="app.handler" key="handler_two" />
+                </service>
+
+                <service id="App\HandlerCollection">
+                    <!-- inject all services tagged with app.handler as first argument -->
+                    <argument type="tagged" tag="app.handler" index-by="key" />
+                </service>
+            </services>
+        </container>
+
+    .. code-block:: php
+
+        // config/services.php
+        use Symfony\Component\DependencyInjection\Argument\TaggedIteratorArgument;
+
+        $container->register(App\Handler\One::class)
+            ->addTag('app.handler', ['key' => 'handler_one']);
+
+        $container->register(App\Handler\Two::class)
+            ->addTag('app.handler', ['key' => 'handler_two']);
+
+        $container->register(App\Handler\HandlerCollection::class)
+            // inject all services tagged with app.handler as first argument
+            ->addArgument(new TaggedIteratorArgument('app.handler', 'key'));
+
+After compilation the ``HandlerCollection`` is able to iterate over your
+application handlers. To retrieve a specific service by it's ``key`` attribute
+from the iterator, we can use ``iterator_to_array`` and retrieve the ``handler_two``:
+to get an array and then retrieve the ``handler_two`` handler :
+
+.. code-block:: php
+
+    // src/Handler/HandlerCollection.php
+    namespace App\Handler;
+
+    class HandlerCollection
+    {
+        public function __construct(iterable $handlers)
+        {
+            $handlers = iterator_to_array($handlers);
+
+            $handler_two = $handlers['handler_two']:
+        }
+    }
+
+
+.. tip::
+
+    You can omit the ``index_attribute_name`` attribute, by implementing a static
```

Comment:
What's this `index_attribute_name` ? It's the first time it's mentioned in the docs.

Reply:
This one was introduced by the new code

Reply:
This doc suggests that there is an option called `index_attribute_name`. I can't see it in the previous examples, where we only use `index_by` or `inex-by`. So, is this an error? Is there a missing example? Am I missing something? Thanks.

Reply:
@Hi @javiereguiluz , 
in fact the option is called ``index_by`` into the ``!tagged argument``. 
With a little example it would be better : 
```yaml
    foo_service:
        class: Foo
        tags:
            - { name: foo,  key: bar_foo}
    foo_service_tagged:
        class: Bar
        arguments: [!tagged { tag: foo, index_by: key  }]
```

What we explain in the documentation, is that we can, here, omit the attribute called ``key`` If we implement a method called ``getDefaultIndexName``

If in the ``index_by`` option I called it `sun` so the omit attribute would be `sun`.

Is that more clear ? 

The rest of the docs explains that with example.

If you find a more intuitive way to explains this, don't hesitate tu push to this PR ;)

Reply:
I agree with Javier, the sentence should be `You can omit the ``key`` attribute...`, then directly show the usage of
```yaml
arguments: [!tagged { tag: 'app.handler', index_by: 'key', default_index_method: 'someFunctionName' }]
```
within the tip, and discard the following.